### PR TITLE
feat: rename table to tableId

### PR DESCRIPTION
**Pull request by alvrs**
- rename `table` to `tableId` wherever it is used as an ID

https://www.notion.so/latticexyz/MUD-naming-conventions-66e11356a1934465b2aa1a951c61a299

### CODE REVIEW

**Comment by alvrs** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1482** <a href="https://app.graphite.dev/github/pr/latticexyz/mud/1482" target="_blank">Graphite</a> 
    * **PR #1484** <a href="https://app.graphite.dev/github/pr/latticexyz/mud/1484" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/latticexyz/mud/1484?utm_source=stack-comment).

**Review by holic — DISMISSED** [review]

File: packages/world/src/modules/keysintable/KeysInTableHook.sol [hunk 1]
```diff
@@ -8,17 +8,17 @@ import { KeysInTable } from "./tables/KeysInTable.sol";
...
   function handleSet(bytes32 tableId, bytes32[] memory key) internal {
     bytes32 keysHash = keccak256(abi.encode(key));
 
-    // If the key has not yet been set in the table...
+    // If the key has not yet been set in the tableId...
     if (!UsedKeysIndex.getHas(tableId, keysHash)) {
       uint40 length = uint40(KeysInTable.lengthKeys0(tableId));
 
-      // Push the key to the list of keys in this table
+      // Push the key to the list of keys in this tableId
```
Comment (dk1a) [inline packages/world/src/modules/keysintable/KeysInTableHook.sol:21]: I think a bunch of comments here (more below) should retain `table`

**Review by dk1a — DISMISSED** [review]

File: packages/store/test/StoreCoreDynamic.t.sol [hunk 2]
```diff
@@ -27,17 +27,17 @@ contract StoreCoreDynamicTest is Test, GasReporter, StoreMock {
...
     uint256 byteLengthToPop,
     FieldLayout fieldLayout
   ) public override {
-    StoreCore.popFromField(table, key, schemaIndex, byteLengthToPop, fieldLayout);
+    StoreCore.popFromField(tableId, key, schemaIndex, byteLengthToPop, fieldLayout);
   }
 
   function setUp() public {
-    // Register table's value schema
+    // Register tableId's value schema
```
Comment (dk1a) [inline packages/store/test/StoreCoreDynamic.t.sol:40]: this one too

**Comment by dk1a** [comment]
I may have missed some, didn't go through every comment

File: packages/store/test/StoreCoreGas.t.sol [hunk 3]
```diff
@@ -77,21 +77,21 @@ contract StoreCoreGasTest is Test, GasReporter, StoreMock {
...
-    bytes32 table = keccak256("some.table");
-    bytes32 table2 = keccak256("other.table");
-    StoreCore.registerTable(table, fieldLayout, defaultKeySchema, valueSchema, new string[](1), new string[](4));
+    bytes32 tableId = keccak256("some.tableId");
+    bytes32 table2 = keccak256("other.tableId");
+    StoreCore.registerTable(tableId, fieldLayout, defaultKeySchema, valueSchema, new string[](1), new string[](4));
 
-    startGasReport("Check for existence of table (existent)");
-    StoreCore.hasTable(table);
+    startGasReport("Check for existence of tableId (existent)");
```
Comment (holic) [inline packages/store/test/StoreCoreGas.t.sol:84]: should this still be "table"?

File: packages/store/test/StoreCoreGas.t.sol [hunk 4]
```diff
@@ -77,21 +77,21 @@ contract StoreCoreGasTest is Test, GasReporter, StoreMock {
       SchemaType.UINT16
     );
     FieldLayout fieldLayout = FieldLayoutEncodeHelper.encode(1, 2, 1, 2, 0);
-    bytes32 table = keccak256("some.table");
-    bytes32 table2 = keccak256("other.table");
-    StoreCore.registerTable(table, fieldLayout, defaultKeySchema, valueSchema, new string[](1), new string[](4));
+    bytes32 tableId = keccak256("some.tableId");
+    bytes32 table2 = keccak256("other.tableId");
```
Comment (holic) [inline packages/store/test/StoreCoreGas.t.sol:81]:
```suggestion
    bytes32 tableId2 = keccak256("other.tableId");
```

File: packages/store/test/StoreCoreGas.t.sol (hunk 3 above)
Comment (holic) [inline packages/store/test/StoreCoreGas.t.sol:84]:
```suggestion
    startGasReport("Check for existence of table (existent)");
```

File: packages/store/test/StoreCoreGas.t.sol [hunk 5]
```diff
@@ -77,21 +77,21 @@ contract StoreCoreGasTest is Test, GasReporter, StoreMock {
...
+    StoreCore.registerTable(tableId, fieldLayout, defaultKeySchema, valueSchema, new string[](1), new string[](4));
 
-    startGasReport("Check for existence of table (existent)");
-    StoreCore.hasTable(table);
+    startGasReport("Check for existence of tableId (existent)");
+    StoreCore.hasTable(tableId);
     endGasReport();
 
-    startGasReport("check for existence of table (non-existent)");
+    startGasReport("check for existence of tableId (non-existent)");
```
Comment (holic) [inline packages/store/test/StoreCoreGas.t.sol:88]:
```suggestion
    startGasReport("check for existence of table (non-existent)");
```

File: packages/store/test/StoreCoreGas.t.sol [hunk 6]
```diff
@@ -103,62 +103,62 @@ contract StoreCoreGasTest is Test, GasReporter, StoreMock {
...
     // Get data
     startGasReport("get static record (1 slot)");
-    StoreCore.getRecord(table, key, fieldLayout);
+    StoreCore.getRecord(tableId, key, fieldLayout);
     endGasReport();
   }
 
   function testSetAndGetStaticDataSpanningWords() public {
-    // Register table
+    // Register tableId
```
Comment (holic) [inline packages/store/test/StoreCoreGas.t.sol:157]:
```suggestion
    // Register table
```

File: packages/store/test/StoreCoreGas.t.sol [hunk 7]
```diff
@@ -170,26 +170,26 @@ contract StoreCoreGasTest is Test, GasReporter, StoreMock {
...
+    StoreCore.getRecord(tableId, key, fieldLayout);
     endGasReport();
   }
 
   function testSetAndGetDynamicData() public {
-    bytes32 table = keccak256("some.table");
+    bytes32 tableId = keccak256("some.tableId");
 
-    // Register table
+    // Register tableId
```
Comment (holic) [inline packages/store/test/StoreCoreGas.t.sol:185]:
```suggestion
    // Register table
```

File: packages/store/test/StoreCoreGas.t.sol [hunk 8]
```diff
@@ -256,17 +256,17 @@ contract StoreCoreGasTest is Test, GasReporter, StoreMock {
   }
 
   function testSetAndGetField() public {
-    bytes32 table = keccak256("some.table");
+    bytes32 tableId = keccak256("some.tableId");
 
-    // Register table
+    // Register tableId
```
Comment (holic) [inline packages/store/test/StoreCoreGas.t.sol:261]:
```suggestion
    // Register table
```

File: packages/store/test/StoreCoreGas.t.sol [hunk 9]
```diff
@@ -634,34 +634,34 @@ contract StoreCoreGasTest is Test, GasReporter, StoreMock {
 
     bytes memory data = abi.encodePacked(bytes16(0x0102030405060708090a0b0c0d0e0f10));
 
-    startGasReport("set record on table with subscriber");
-    StoreCore.setRecord(table, key, data, fieldLayout);
+    startGasReport("set record on tableId with subscriber");
```
Comment (holic) [inline packages/store/test/StoreCoreGas.t.sol:637]:
```suggestion
    startGasReport("set record on table with subscriber");
```

File: packages/store/test/StoreCoreGas.t.sol [hunk 10]
```diff
@@ -634,34 +634,34 @@ contract StoreCoreGasTest is Test, GasReporter, StoreMock {
...
+    StoreCore.setRecord(tableId, key, data, fieldLayout);
     endGasReport();
 
     data = abi.encodePacked(bytes16(0x1112131415161718191a1b1c1d1e1f20));
 
-    startGasReport("set static field on table with subscriber");
-    StoreCore.setField(table, key, 0, data, fieldLayout);
+    startGasReport("set static field on tableId with subscriber");
```
Comment (holic) [inline packages/store/test/StoreCoreGas.t.sol:643]:
```suggestion
    startGasReport("set static field on table with subscriber");
```

File: packages/world/src/modules/keysintable/KeysInTableHook.sol [hunk 11]
```diff
@@ -8,7 +8,7 @@ import { KeysInTable } from "./tables/KeysInTable.sol";
 import { UsedKeysIndex } from "./tables/UsedKeysIndex.sol";
 
 /**
- * Note: if a table with composite keys is used, only the first key is indexed
+ * Note: if a tableId with composite keys is used, only the first key is indexed
```
Comment (holic) [inline packages/world/src/modules/keysintable/KeysInTableHook.sol:11]:
```suggestion
 * Note: if a table with composite keys is used, only the first key is indexed
```

**Pull request merged**